### PR TITLE
fix: synchronous component rendering + re-add style logic + upgrade vite

### PR DESCRIPTION
**Pull request by AndrewLeedham**
- Removed the `Promise.all` component rendering, which was causing vite to fall over with dependency resolution.
- Upgraded vite to fix this issue: https://github.com/vitejs/vite/issues/5061.
- Re-added the style logic to add inline `<style>` tags back to component renders.

Merged the babel/esbuild changes from main.

### CODE REVIEW

**Comment by AndrewLeedham** [comment]
@bholmesdev Apart from testing this should bring the fix back to functional parity with main.

File: src/plugin/applyViteHtmlTransform.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ const ssrRegex = RegExp(toSSRComment('([0-9]+)'), 'g')
  * @property {import('./componentAttrStore').ComponentAttrStore} componentAttrStore
  * @param {ApplyViteHtmlTransformParams}
  * @param {import('.').SlinkityConfigOptions}
- * @returns {string} - HTML with statically rendered content and Vite transforms applied
+ * @returns {Promise<string>} - HTML with statically rendered content and Vite transforms applied
```
Comment (Holben888) [inline src/plugin/applyViteHtmlTransform.js:17]: Ah, thanks for fixing that type 👍 Could really use an eslint plugin to flag bad TS docs...

**Review by Holben888 — APPROVED** [review]
No complaints, exactly what I was hoping for. Let's go!

File: src/plugin/applyViteHtmlTransform.js (hunk 1 above)
Comment (AndrewLeedham) [reply src/plugin/applyViteHtmlTransform.js:17]: https://www.npmjs.com/package/eslint-plugin-jsdoc might be what you're looking for

**Pull request merged**